kubetest/gke_test.go

```diff
@@ -0,0 +1,174 @@
+		"project",
+		"",
+		"region",
+		"network",
+		"image",
+		"imageFamily",
+		"imageProject",
+		"cluster",
+		"sshProxyInstanceName",
+		new(string),
+		new(string),
+	)
+}
+
+func TestParseInstanceGroupsFromGCloud(t *testing.T) {
+	urlTemplate := `https://www.googleapis.com/compute/v1/projects/test-project/zones/%s/instanceGroupManagers/%s-io-default-pool-6f8ec327-grp`
+	zoneA := "us-central1-a"
+	zoneB := "us-central1-b"
+	zoneC := "us-central1-c"
+	urlZoneADefault := fmt.Sprintf(urlTemplate, zoneA, "gke")
+	urlZoneBDefault := fmt.Sprintf(urlTemplate, zoneB, "gke")
+	urlZoneCDefault := fmt.Sprintf(urlTemplate, zoneC, "gke")
+	urlZoneATest := fmt.Sprintf(urlTemplate, zoneA, "test")
+	urlZoneBTest := fmt.Sprintf(urlTemplate, zoneB, "test")
+	urlZoneCTest := fmt.Sprintf(urlTemplate, zoneC, "test")
+
+	expectedPathADefault := `zones/us-central1-a/instanceGroupManagers/gke-io-default-pool-6f8ec327-grp`
+	expectedPathBDefault := `zones/us-central1-b/instanceGroupManagers/gke-io-default-pool-6f8ec327-grp`
+	expectedPathCDefault := `zones/us-central1-c/instanceGroupManagers/gke-io-default-pool-6f8ec327-grp`
+	expectedPathATest := `zones/us-central1-a/instanceGroupManagers/test-io-default-pool-6f8ec327-grp`
+	expectedPathBTest := `zones/us-central1-b/instanceGroupManagers/test-io-default-pool-6f8ec327-grp`
+	expectedPathCTest := `zones/us-central1-c/instanceGroupManagers/test-io-default-pool-6f8ec327-grp`
+	expectedNameDefault := `gke-io-default-pool-6f8ec327-grp`
+	expectedNameTest := `test-io-default-pool-6f8ec327-grp`
+	expectedUniq := `6f8ec327`
+
+	cases := []struct {
+		input                 string
+		instance_group_prefix string
+		expected              []*ig
+	}{
+		// Zonal
+		{
+			input:                 urlZoneADefault,
+			instance_group_prefix: "",
+			expected: []*ig{
+				{
+					path: expectedPathADefault,
+					zone: zoneA,
+					name: expectedNameDefault,
+					uniq: expectedUniq,
+				},
+			},
+		},
+		{
+			input:                 urlZoneADefault,
+			instance_group_prefix: "test",
+			expected:              nil,
+		},
+		{
+			input:                 urlZoneATest,
+			instance_group_prefix: "test",
+			expected: []*ig{
+				{
+					path: expectedPathATest,
+					zone: zoneA,
+					name: expectedNameTest,
+					uniq: expectedUniq,
+				},
+			},
+		},
+		// Regional
+		{
+			input:                 strings.Join([]string{urlZoneADefault, urlZoneBDefault, urlZoneCDefault}, ";"),
+			instance_group_prefix: "",
+			expected: []*ig{
+				{
+					path: expectedPathADefault,
+					zone: zoneA,
+					name: expectedNameDefault,
+					uniq: expectedUniq,
+				},
+				{
+					path: expectedPathBDefault,
+					zone: zoneB,
+					name: expectedNameDefault,
+					uniq: expectedUniq,
+				},
+				{
+					path: expectedPathCDefault,
+					zone: zoneC,
+					name: expectedNameDefault,
+					uniq: expectedUniq,
+				},
+			},
+		},
+		{
+			input:                 strings.Join([]string{urlZoneADefault, urlZoneBDefault, urlZoneCDefault}, ";"),
```

Comment:
nit: that could be zonal - same test, less code

Reply:
Zonal is also implemented in line 92. I guess I could remove this test.